Update Kotlin Version

k163377 opened the pull request:
Kotlin 1.5 has been deprecated and the minimum supported Kotlin version will be updated to 1.6 for Jackson 2.16 and later.
https://kotlinlang.org/docs/kotlin-evolution.html#compatibility-flags

cowtowncoder commented:
Makes sense: should also mention this in README, I think? (supported versions section; in Jackson / Kotlin version matrix)

k163377 commented:
@cowtowncoder 
Thank you for updating the Wiki.
I have also added a note regarding the destructive change on #617.
https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.16#kotlin

> Makes sense: should also mention this in README

I thought I changed it below, but am I forgetting something?
https://github.com/FasterXML/jackson-module-kotlin/pull/684/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

cowtowncoder commented:
@k163377 No, I just missed the update myself. Looks good, thanks!
